### PR TITLE
Feature/hl poi gravitation

### PR DESCRIPTION
**Pull request by HLucasTargomo**
- Add POI Gravitation endpoint

### CODE REVIEW

File: src/main/java/com/targomo/client/api/json/DefaultSourceCoordinateMapSerializer.java [hunk 1]
```diff
@@ -20,12 +20,20 @@ public void serialize(Object o, JsonGenerator jsonGenerator, SerializerProvider
...
-            jsonGenerator.writeEndObject(); // {
+            if( entry.getValue().getAggregationInputParameters() != null){
+                jsonGenerator.writeFieldName("aggregationInputParameters");
+                jsonGenerator.writeStartObject();
+                jsonGenerator.writeNumberField("inputFactor", entry.getValue().getAggregationInputParameters().getInputFactor());
+                jsonGenerator.writeNumberField("gravitationAttractionStrength", entry.getValue().getAggregationInputParameters().getGravitationAttractionStrength());
+                jsonGenerator.writeBooleanField("gravitationPositiveInfluence", entry.getValue().getAggregationInputParameters().getGravitationPositiveInfluence());
+                jsonGenerator.writeEndObject();
+            }
+            jsonGenerator.writeEndObject();
```
Comment (GideonTargomo) [inline src/main/java/com/targomo/client/api/json/DefaultSourceCoordinateMapSerializer.java:36]: I think you need to add this to the `DefaultSourceGeometriesMapSerializer` too.

Comment (HLucasTargomo) [reply src/main/java/com/targomo/client/api/json/DefaultSourceCoordinateMapSerializer.java:35]: oh yes thanks! good catch!

File: README.md [hunk 2]
```diff
@@ -52,7 +52,7 @@ nexus and to the maven repo (last is only possibly from master).
 ## Release Notes
 
 ### 0.3.0
-- To fill
+- Add POI Gravitation endpoint
```
Comment (DavidRedlich) [inline README.md:55]: this seems to also include that you can use geometries as source? is this an extra MR that should be reviewed before or together? if together can you please add this to the release notes?

File: src/main/java/com/targomo/client/api/geo/AbstractLocation.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
+package com.targomo.client.api.geo;
+
+import com.targomo.client.api.pojo.AggregationInputParameters;
+
+public abstract class AbstractLocation {
+
+    protected String id;
+    protected AggregationInputParameters aggregationInputParameters;
```
Comment (DavidRedlich) [inline src/main/java/com/targomo/client/api/geo/AbstractLocation.java:8]: adding this to the abstract location could be a little confusing tbh. in multigraph AggregationInputParameters are at a different location. We might have to look for an alternative way although I don't really have an idea how we could do it better. We probably should have a quick brainstorming session or chat about how we model this here.

File: README.md (hunk 2 above)
Comment (HLucasTargomo) [reply README.md:55]: No, using geometries as source was already included before

File: src/main/java/com/targomo/client/api/geo/AbstractLocation.java (hunk 3 above)
Comment (HLucasTargomo) [reply src/main/java/com/targomo/client/api/geo/AbstractLocation.java:8]: hum ok how and when should we chat about that?

File: src/main/java/com/targomo/client/api/pojo/LocationProperties.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package com.targomo.client.api.pojo;
+
+import lombok.*;
+
+
+// Only relevant for POI service's gravitation requests
```
Comment (atomicbomberman) [inline src/main/java/com/targomo/client/api/pojo/LocationProperties.java:6]: If its only relevant for POI service requests why add it to the root class of locations. Would it make sense to rather create a new sub-class "PoiSourceCoordinate" or something?

**Review by atomicbomberman — CHANGES_REQUESTED** [review]

**Review by atomicbomberman — APPROVED** [review]

**Pull request merged**